### PR TITLE
Fix #36 - port to QueueClassic 3.1's scheduling; drop queue_classic-later

### LINKED ISSUE #36: Get ready for queue_classic 3.1 and it's build-in scheduling
This gem is relying on queue_classic 3.0 and if you have queue_classic-later, it will work with it.

What does it do? It's looking at the queue_classic-later table when you click on the "Scheduled Jobs" link at the top. With queue_classic 3.1 though, there is now built-in scheduling similar to what queue_classic-later did. That said, it has a different approach. All jobs are now in the `queue_classic_jobs` table. There is no table for scheduling anymore. We want to get the queue_classic_admin aligned with that. We want only the job ready to show by default and the "Scheduled Jobs" link should still show jobs that are scheduled for later.

You can get additional context in https://github.com/QueueClassic/queue_classic/pull/217 and https://github.com/QueueClassic/queue_classic/issues/228. Those PR in other queue_classic related project might also help a bit, maybe not either: https://github.com/rainforestapp/queue_classic_plus/pull/4 and https://github.com/rainforestapp/queue_classic_matchers/pull/1.

### PR DESCRIPTION
**Pull request by mnp**
Please review and comment.

This is mostly a bulk rename to keep terminology consistent.  All mention of the "later" concept has been replaced by "scheduled".  This aligns with QC 3.1's scheduled_at column name.

Question about more terminology: QC doesn't seem to make a distinction between enqueued_at and scheduled_at.  In terms of lifecycle, I would expect it go: scheduled to be enqueued, enqueued for work, locked, removed. Is that conception inaccurate? The UI seems to imply it but the behavior does not.

Thanks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,16 +2,18 @@
 
 [![Build Status](https://travis-ci.org/QueueClassic/queue_classic_admin.png)](https://travis-ci.org/QueueClassic/queue_classic_admin)
 
-An admin interface for the [queue_classic](https://github.com/ryandotsmith/queue_classic) and [queue_classic-later](https://github.com/dpiddy/queue_classic-later) gems.
+An admin interface for the [queue_classic](https://github.com/ryandotsmith/queue_classic) gem.
 
 **IMPORTANT: this branch is for QueueClassic 3. See the [queueclassic-2-support](https://github.com/dpiddy/queue_classic-later/tree/queueclassic-2-support) branch for prior version support.**
```
Comment (jipiboily) [inline README.md]: Please update this with a message saying that this is 3.1 and not, and add a message saying where to find qc 3 support, like a branch named "queueclassic-3-support", I'll create it later.

File: app/helpers/queue_classic_admin/application_helper.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 module QueueClassicAdmin
   module ApplicationHelper
-    def has_qc_later?
-      QC.respond_to?(:enqueue_in) &&
-        defined?(QC::Later) &&
-        ActiveRecord::Base.connection.table_exists?(QC::Later::TABLE_NAME)
+
+    def has_qc_3_1?
```
Comment (jipiboily) [inline app/helpers/queue_classic_admin/application_helper.rb]: I would personally prefer a name reflecting what we're actually doing, which is looking for support of scheduling. Something like `has_scheduling?` would make more sense to me. Thoughts?

File: app/models/queue_classic_admin/queue_classic_scheduled_job.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module QueueClassicAdmin
+  class QueueClassicScheduledJob < ActiveRecord::Base
```
Comment (jipiboily) [inline app/models/queue_classic_admin/queue_classic_scheduled_job.rb]: Not sure to understand what this model is used for...as it's going to do the exact same thing as the `QueueClassicJob` model, no?

File: app/helpers/queue_classic_admin/application_helper.rb (hunk 2 above)
Comment (mnp) [inline app/helpers/queue_classic_admin/application_helper.rb]: Yes, that would be more consistent.

File: spec/dummy/db/schema.rb [hunk 4]
```diff
@@ -27,12 +27,11 @@
 
   add_index "queue_classic_jobs", ["q_name", "id"], name: "idx_qc_on_name_only_unlocked", where: "(locked_at IS NULL)", using: :btree
 
-  create_table "queue_classic_later_jobs", force: true do |t|
-    t.string   "q_name"
-    t.string   "method"
+  create_table "queue_classic_later_jobs", id: false, force: :cascade do |t|
```
Comment (jipiboily) [inline spec/dummy/db/schema.rb]: This should not be here anymore, not trace of `queue_classic_later_jobs` should be left I think.

**Comment by jipiboily** [comment]
@mnp Thanks for the PR! I added a few notes here and there.

There are at least two things I would love to see before I can merge this:
- `QueueClassicScheduledJob` seems to return all jobs, even the ones that are not scheduled for later. Am I wrong? Did I miss something or right now it will do the same thing as `QueueClassicJob`?
- The build is failing with a bunch of failures :grinning:

File: spec/dummy/db/schema.rb (hunk 4 above)
Comment (mnp) [inline spec/dummy/db/schema.rb]: Yes, updating schema.  Also having that debate with Travis right now :)

On Mon, Jan 26, 2015 at 3:09 PM, Jean-Philippe Boily <
notifications@github.com> wrote:

> In spec/dummy/db/schema.rb
> <https://github.com/QueueClassic/queue_classic_admin/pull/37#discussion_r23560240>
> :
>
> > @@ -27,12 +27,11 @@
> >
> >    add_index "queue_classic_jobs", ["q_name", "id"], name: "idx_qc_on_name_only_unlocked", where: "(locked_at IS NULL)", using: :btree
> >
> > -  create_table "queue_classic_later_jobs", force: true do |t|
> > -    t.string   "q_name"
> > -    t.string   "method"
> > +  create_table "queue_classic_later_jobs", id: false, force: :cascade do |t|
>
> This should not be here anymore, not trace of queue_classic_later_jobs
> should be left I think.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/QueueClassic/queue_classic_admin/pull/37/files#r23560240>
> .
>



-- 
Mitchell Perilstein <mitchell.perilstein@gmail.com>
(484) 876-1444

**Comment by jipiboily** [comment]
@mnp are you planning to address the comment? Thanks!

**Pull request closed without merging**